sled-agent/src/bootstrap/http_entrypoints.rs

```diff
@@ -0,0 +1,46 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! HTTP entrypoint functions for the bootstrap agent's API.
+//!
+//! Note that the bootstrap agent also communicates over Sprockets,
+//! and has a separate interface for establishing the trust quorum.
+
+use crate::bootstrap::agent::Agent;
+use crate::updates::Component;
+use dropshot::{
+    endpoint, ApiDescription, HttpError, HttpResponseOk, RequestContext,
+};
+use omicron_common::api::external::Error;
+use std::sync::Arc;
+
+type BootstrapApiDescription = ApiDescription<Arc<Agent>>;
+
+/// Returns a description of the bootstrap agent API
+pub(crate) fn api() -> BootstrapApiDescription {
+    fn register_endpoints(
+        api: &mut BootstrapApiDescription,
+    ) -> Result<(), String> {
+        api.register(components_get)?;
+        Ok(())
+    }
+
+    let mut api = BootstrapApiDescription::new();
+    if let Err(err) = register_endpoints(&mut api) {
+        panic!("failed to register entrypoints: {}", err);
+    }
+    api
+}
+
+#[endpoint {
```

Comment:
I know this is a draft but it would be helpful to mention here the (relatively narrow, as I understand it) set of situations where this endpoint is meant to be called.

Reply:
Sure, I can add some notes before this is merged. This endpoint is intended to be used by wicket, as part of the recovery / MUP process, to scan for host software information.

I also plan on exposing a similar endpoint out of the "normal" sled agent interface, which will use the same implementation under-the-hood.

However, I'm pushing this out of the bootstrap agent for now because of the reasons outlined in #2215  - we need to access this interface before trust quorum has been established, which means the sled agent isn't actually listening on an underlay address when we first want wicket to query this interface.

Reply:
Doc comment added